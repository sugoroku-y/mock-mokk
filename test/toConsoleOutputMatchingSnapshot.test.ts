import './toConsoleOutputMatchingSnapshot';
import './toFailMatchingSnapshot';

describe('.toConsoleOutputMatchingSnapshot', () => {
  test.each`
    name         | pass
    ${undefined} | ${true}
    ${'log'}     | ${true}
    ${'debug'}   | ${true}
    ${'info'}    | ${true}
    ${'warn'}    | ${true}
    ${'error'}   | ${true}
    ${undefined} | ${false}
    ${'log'}     | ${false}
    ${'debug'}   | ${false}
    ${'info'}    | ${false}
    ${'warn'}    | ${false}
    ${'error'}   | ${false}
  `(
    '$name $pass',
    ({
      name,
      pass,
    }: {
      name: 'log' | 'debug' | 'info' | 'warn' | 'error' | undefined;
      pass: boolean;
    }) => {
      if (pass) {
        expect(() =>
          console[name ?? 'log'](`test: ${name} ${pass}`),
        ).toConsoleOutputMatchingSnapshot(name);
      } else {
        expect(() => {
          expect(() =>
            console[name ?? 'log'](`test: ${name} ${pass}:x`),
          ).toConsoleOutputMatchingSnapshot(name);
        }).toFailMatchingSnapshot();
      }
    },
  );
});
