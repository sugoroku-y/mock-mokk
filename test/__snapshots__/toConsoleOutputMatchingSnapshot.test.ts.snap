// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toConsoleOutputMatchingSnapshot debug false: toConsoleOutputMatchingSnapshot:debug 1`] = `
Array [
  Array [
    "test: debug false",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot debug true: toConsoleOutputMatchingSnapshot:debug 1`] = `
Array [
  Array [
    "test: debug true",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot error false: toConsoleOutputMatchingSnapshot:error 1`] = `
Array [
  Array [
    "test: error false",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot error true: toConsoleOutputMatchingSnapshot:error 1`] = `
Array [
  Array [
    "test: error true",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot info false: toConsoleOutputMatchingSnapshot:info 1`] = `
Array [
  Array [
    "test: info false",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot info true: toConsoleOutputMatchingSnapshot:info 1`] = `
Array [
  Array [
    "test: info true",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot log false: toConsoleOutputMatchingSnapshot:log 1`] = `
Array [
  Array [
    "test: log false",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot log true: toConsoleOutputMatchingSnapshot:log 1`] = `
Array [
  Array [
    "test: log true",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot undefined false: toConsoleOutputMatchingSnapshot:log 1`] = `
Array [
  Array [
    "test: undefined false",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot undefined true: toConsoleOutputMatchingSnapshot:log 1`] = `
Array [
  Array [
    "test: undefined true",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot warn false: toConsoleOutputMatchingSnapshot:warn 1`] = `
Array [
  Array [
    "test: warn false",
  ],
]
`;

exports[`.toConsoleOutputMatchingSnapshot warn true: toConsoleOutputMatchingSnapshot:warn 1`] = `
Array [
  Array [
    "test: warn true",
  ],
]
`;
